base/compiler/tfuncs.jl

```diff
@@ -902,9 +927,9 @@ function fieldtype_tfunc(@nospecialize(s0), @nospecialize(name))
         ft = ftypes[fld]
     end
 
-    exact = (isa(s0, Const) || isType(s0)) && !has_free_typevars(s)
+    exactft = exact || !has_free_typevars(ft)
```

Comment:
Is `exact` enough here? What about types that are `==`, but not `===`?

Reply:
for `Const` I mean. I do agree that we can at least say `Type{ft}` with this condition.

Reply:
Good point. I think `Const` is actually almost always the wrong answer here